gobblin-runtime/src/main/java/gobblin/runtime/TaskContext.java

```diff
@@ -132,6 +132,9 @@ public WriterOutputFormat getWriterOutputFormat(int branches, int index) {
   public List<Converter<?, ?, ?, ?>> getConverters(int index) {
     String converterClassKey =
         ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.CONVERTER_CLASSES_KEY, index);
+
+    this.workUnitState.setProp(ConfigurationKeys.FORK_BRANCH_ID_KEY, index);
```

Comment:
If you are talking about converters created in each `Fork` (see the member variable `converter` in `Fork`), then you can add it to the `Fork` constructor before the initialization of `converter`.